src/storage/observable/fromTask.ts

```diff
@@ -0,0 +1,14 @@
+import { storage } from 'firebase/app';
+import 'firebase/storage';
+import { Observable } from 'rxjs/Observable';
+
+export function fromTask(task: storage.UploadTask) {
+  return new Observable<storage.UploadTaskSnapshot | undefined>(subscriber => {
```

Comment:
I'm thinking we should at least `share()` this, perhaps even consider `share(1)`. I forsee lots of tickets opened if we don't.

Also I haven't run + dug in but is this task lazy? E.g, won't start uploading until subscribed to? Will it double upload if cold? That may violate the expectations of beginners. In which case we should `.share()` and then `subscribe()` in here.

Reply:
Good call. I'll check that and add tests.

Reply:
You don't need to share this, since you're closing over the task, the task is effectively a Subject, since it multicasts anyhow.